### PR TITLE
fix: cursor pagination

### PR DESCRIPTION
**Pull request by BrooksFlannery**
Continued from @Nyan-Prakash 's #661 

---
## Summary by cubic
Fixes cursor-based pagination by correctly parsing GET query params and switching to deterministic keyset pagination on (createdAt, id). Adds strict validation, legacy cursor support, and error telemetry, with comprehensive tests.

- **Bug Fixes**
  - Parse URL query params with searchParamsToObject and merge into tRPC input on GET.
  - Validate limit/cursor (Zod), enforce single values, limit <= 100, and return 400 with telemetry on invalid params. Accept legacy cursor (no id) with a deprecation warning.
  - Implement stable keyset pagination in createPaginatedSelectFunction using (createdAt, id), exclude anchor row, and include id/direction in nextCursor. Supports forward/backward and correct hasMore/total.

- **Tests**
  - New tests for searchParamsToObject and pagination parsing/validation, including legacy cursor.
  - Extensive pagination tests covering ordering, boundaries, duplicate timestamps, forward/backward, limit edge cases, and cursor continuity.

<sup>Written for commit 4e0436b. Summary will update automatically on new commits.</sup>

### CODE REVIEW

**Review by angihe93 — CHANGES_REQUESTED** [review]

File: platform/flowglad-next/src/app/api/v1/[...path]/route.ts [hunk 1]
```diff
@@ -349,28 +356,59 @@ const innerHandler = async (
...
+
+            return NextResponse.json(
+              { error: (error as Error).message },
+              { status: 400 }
+            )
+          }
         }
+
+        let newReq: Request
+
```
Comment (agreea) [inline platform/flowglad-next/src/app/api/v1/[...path]/route.ts]: can we add some code comments explaining what we're doing here and why? ts-doc format to be consistent with the rest of the file please

**Review by agreea — APPROVED** [review]
Good to ship pending a few small changes:
- code comments
- consolidate some of the test cases into single test case
- resolution of the cubic comment here: https://github.com/flowglad/flowglad/pull/680#discussion_r2471634088

**Review by angihe93 — APPROVED** [review]
